### ISSUE TITLE
Mono bug calling methods

### ISSUE THREAD

**Issue by splitice** [davideicardi/DynamicExpresso#34]
When attempting to use code which calls ```int.Parse(var)``` an exception is thrown in mono. This code works fine in .NET.

The error is as follows:
```DynamicExpresso.NoApplicableMethodException: No applicable method 'Parse' exists in type 'Int32' (at index 4)```

**Comment by splitice** [comment]
A bit of fiddeling later and I realized its a behaviour difference regarding String casting.

var in this case is an object type which needs .ToString()

**Comment by davideicardi** [comment]
Can you give me a full Dynamic Expresso example (or better a unit test) to reproduce it?
What version of Mono are you using?

Int32.Parse should accept a string and as I know also with .NET it is not converted automatically (implicit cast).

thanks!

**Comment by splitice** [comment]
I did some more debugging, I would say although its an inconsistency - its probably .NET that is actually incorrect. I am passing in a ```Dictionary<String,object>```, with say one member of type string (or something that has ToString()). I had attempted to just do an Expression like ```int.Parse(dict["key"])>100```.

This worked in .NET when I did the unit tests locally, failed on Travis-CI on mono.

This worked in .NET but in Mono it failed (with that cryptic error message, when the fault was actually type safety).

I am now working on an automatically casting dictionary, seems like the best approach for now. Thanks.
